src/queue.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+export default class Queue {
+  constructor() {
+    let _items = []
+    let _length = 0
+
+    this.getItems = function() {
+      return _items
+    }
+    this.setItems = function( newItems ) {
+      _items = newItems
+    }
+
+    this.getLength = function() {
+      return _length
+    }
+    this.incrementLength = function() {
+      _length++
+    }
+    this.decrementLength = function() {
+      _length--
+    }
+  }
+
+  enqueue( item ) {
+    let items = this.getItems()
+    let index = this.getLength()
+    items[ index ] = item
+    this.incrementLength()
+    this.setItems( items )
+  }
+
+  dequeue() {
+    if ( this.isEmpty() ) {
+      return null
+    }
+    let items = this.getItems()
+    let target = items[ 0 ]
+
+    delete items[ 0 ]
```

Comment:
you could also pop. or splice. or delete. Which one is better/faster? I'm not sure.